certbot/certbot/ari.py

```diff
@@ -0,0 +1,64 @@
+from cryptography.hazmat.primitives import hashes
+from cryptography import x509
+from cryptography.x509 import ocsp
+import josepy as jose
+import requests
+
+from acme.messages import RenewalInfo
+from certbot.interfaces import RenewableCert  # pylint: disable=unused-import
+
+class AriChecker(object):
+    """This class checks ACME Renewal Info."""
+
+    def __init__(self, ari_endpoint: string) -> None:
+        self.ari_endpoint = ari_endpoint.rstrip('/')
+
+    def _compute_path(self, cert: RenewableCert, issuer: RenewableCert) -> bool:
+        # Rather than compute the serial, issuer key hash, and issuer name hash
+        # ourselves, we instead build an OCSP Request and extract those fields.
+        builder = ocsp.OCSPRequestBuilder()
+        builder = builder.add_certificate(cert, issuer, hashes.SHA1())
+        ocspRequest = builder.build()
+
+        # Construct the ARI path from the OCSP CertID sequence.
+        key_hash = ocspRequest.issuer_key_hash.hex()
+        name_hash = ocspRequest.issuer_name_hash.hex()
+        serial = hex(ocspRequest.serial_number)[2:]
+        path = f"{key_hash}/{name_hash}/{serial}"
+
+        return '/'.join(self.ari_endpoint, path)
```

Comment:
If I'm reading the latest RFC correctly, it looks like the full CertID DER now goes in the URL. Would it be helpful if pyca/cryptography had an API for that?